cmd/notation/login.go

```diff
@@ -117,12 +136,41 @@ func readPassword(opts *loginOpts) error {
 	return nil
 }
 
-func readLine() (string, error) {
-	passwordBytes, err := io.ReadAll(os.Stdin)
+func readLine(r io.Reader) (string, error) {
+	passwordBytes, err := io.ReadAll(r)
 	if err != nil {
 		return "", err
 	}
 	password := strings.TrimSuffix(string(passwordBytes), "\n")
 	password = strings.TrimSuffix(password, "\r")
 	return password, nil
 }
+
+func readUsernameFromPrompt(reader *bufio.Reader) (string, error) {
+	fmt.Print("Username: ")
+	username, err := reader.ReadString('\n')
+	if err != nil {
+		return "", fmt.Errorf("error reading username: %w", err)
+	}
+	username = strings.TrimSpace(username)
+	return username, nil
+}
+
+func readPasswordFromPrompt(reader *bufio.Reader) (string, error) {
+	fmt.Print("Password: ")
+	if term.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
nit: There is an edge case but we can fix later. It is possible that the password has been read by the `bufio.Reader` so that `os.Stdin` has nothing to read. This edge case can only be possibly triggered by the following steps

1. Open a text editor and type username and password there.
2. Open a terminal and run `notation login`.
3. Copy the username and password from the text editor to the terminal and hit enter
4. `notation` will ask the user to type the password again, which is not expected.

Reply:
Oh I just tried your steps with pasting the username and password from text editor, and it works for me. The login succeeded.